lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,459 @@
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    def _update_metadata(self, cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "lightning_probability"
+        2. Remove "threshold" coord
+             (or causes iris.exceptions.CoordinateNotFoundError)
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
+        return new_cube
+
+    def _modify_first_guess(self, cube, first_guess_lightning_cube,
+                            lightning_rate_cube, prob_precip_cube,
+                            prob_vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            first_guess_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube (closest slice in
+                time is used with a maximum time mismatch of 2 hours).
+                This is included to allow this cube to come from a different
+                modelling system, such as the UM.
+
+            lightning_rate_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            prob_precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
+                Must have same other dimensions as cube
+
+            prob_vii_cube (iris.cube.Cube):
+                Radar-derived vertically integrated ice content (VII)
+                Must have same x and y dimensions as cube
+                Time should be a scalar coordinate
+                Must have a threshold coordinate with points matching
+                self.vii_thresholds
+                Can be <No cube> or None or anything that evaluates to False
+
+        Returns:
+            new_prob_lightning_cube (iris.cube.Cube):
+                Output cube containing Nowcast lightning probability.
+
+        Raises:
+            iris.exceptions.ConstraintMismatchError:
+                If lightning_rate_cube or first_guess_lightning_cube do not
+                contain the expected times.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over('time'):
```

Comment:
I'm going to suggest a few inline comments for this function which might help the unwary developer.  Feel free to disagree.  For here, I'd say, `Loop over required forecast validity times`.